subsys/zbus/zbus_runtime_observers.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+struct zbus_observer_node {
+	sys_snode_t node;
+	struct zbus_observer *obs;
+};
+
+static struct zbus_observer_node
+	_zbus_runtime_obs_list[CONFIG_ZBUS_RUNTIME_OBSERVERS_FREELIST_SIZE];
```

Comment:
What is the rationale for keeping this in a separate buffer? Why not conditionally add the `sys_node_t node` to the `struct zbus_observer` if `CONFIG_ZBUS_RUNTIME_OBSERVERS_REGISTRATION` is set (like you do for the channel)?